qiime2/core/tests/test_cache.py

```diff
@@ -158,6 +158,48 @@ def test_is_not_cache(self):
         """
         self.assertFalse(Cache.is_cache(self.not_cache_path))
 
+    def test_file_is_not_cache(self):
+        """Verfies that a single file is not a cache
+        """
+        # Create a thing and make sure it exists and is a file
+        x = tempfile.NamedTemporaryFile()
+        os.path.isfile(x.name)
+
+        # Make sure that file is not a cache
+        self.assertFalse(Cache.is_cache(x.name))
+
+    def test_thing_that_does_not_exist_is_not_cache(self):
+        """A thing that doesn't even exist on the filesystem is not a cache
+        """
+        # Create a random string that isn't on the filesystem at all
+        tries = 1
+
+        random_string = ''.join(
+            random.choices(
+                string.ascii_lowercase +
+                string.ascii_uppercase +
+                string.digits,
+                k=10))
+
+        while os.path.exists(random_string):
+            if tries == 10:
+                raise ValueError(
+                    "We randomly generated 10 alphanumeric strings 10 "
+                    "characters long and somehow all 10 existed as paths on "
+                    "your filesystem.")
```

Comment:
This is guaranteed to be raised if you run this in Chloe's or Andrew's downloads folder

Reply:
@cherman2

Reply:
Not nice!